Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1

```diff
@@ -250,50 +250,58 @@ Set-Alias -Scope Global Rebuild-Solution Invoke-RebuildSolution
         if (-not [string]::IsNullOrWhiteSpace($env:msbuild_buildConfiguration)) { $buildConfiguration = $env:msbuild_buildConfiguration }
 
-        $msBuildParameters = @(
-            "`"/property:Configuration=$buildConfiguration`"",
-            "`"/target:$target`"",
-            "`"/verbosity:$verbosity`"",
-            "`"/nr:$noReuse`"",
-            "`"/p:Platform=$platformEnvironmentVariable`""
-        )
-
-        if (-not [string]::IsNullOrWhiteSpace($maxCpuCount)) {
-            $msBuildParameters += "`"/m:$maxCpuCount`""
-        }
-        else {
-            $msBuildParameters += "`"/m`""
-        }
-
         $params = @{
             Path                           = $solutionPath
-            MsBuildParameters              = ($msBuildParameters -join ' ')
-            ShowBuildOutputInCurrentWindow = $true
+            BuildConfiguration             = $buildConfiguration
             LogVerbosityLevel              = $verbosity
         }
 
-        if (-not [string]::IsNullOrWhiteSpace($msBuildFilePath)) { $params.MsBuildFilePath = $msBuildFilePath }
-
         ($params).GetEnumerator() | Sort-Object -Property Name | Format-Table -HideTableHeaders -AutoSize -Wrap | Out-Host
-        $buildResult = Invoke-MsBuild @params
 
-        if ($null -eq $buildResult.BuildSucceeded) { throw "Unsure if build passed or failed: $($buildResult.Message)" }
-        if ($buildResult.BuildSucceeded -eq $true) { return }
-        if ($buildResult.BuildSucceeded -eq $false) {
+		# Local Variables.
+		$BuildLogDirectoryPath = $env:Temp
+		
+		# Always get the full path to the Log files directory.
+		$BuildLogDirectoryPath = [System.IO.Path]::GetFullPath($BuildLogDirectoryPath)
+		
+		# Local Variables.
+		$solutionFileName = (Get-ItemProperty -LiteralPath $solutionPath).Name
+		$buildLogFilePath = (Join-Path -Path $BuildLogDirectoryPath -ChildPath $solutionFileName) + ".msbuild.log"
+		$buildErrorsLogFilePath = (Join-Path -Path $BuildLogDirectoryPath -ChildPath $solutionFileName) + ".msbuild.errors.log"
+		$buildSucceeded = $null
+        
+		# Build
+		dotnet build $solutionPath -c $buildConfiguration -v $verbosity /flp:v=$verbosity /flp:logfile=$buildLogFilePath
+		
+		# If we can't find the build's log file in order to inspect it, write a warning and return null.
+		if (!(Test-Path -LiteralPath $buildLogFilePath -PathType Leaf))
+		{
+			$buildSucceeded = $null
+			
+			Write-Warning ("Cannot find the build log file at '$buildLogFilePath', so unable to determine if build succeeded or not.")
+		}
+		
+		# Get if the build succeeded or not.
+		[bool] $buildOutputDoesNotContainFailureMessage = $null -eq (Select-String -Path $buildLogFilePath -Pattern "Build FAILED." -SimpleMatch)
+		$buildSucceeded = $buildOutputDoesNotContainFailureMessage
+
+        if ($null -eq $buildSucceeded) { throw "Unsure if build passed or failed: FAILED to build ""$solutionPath"". Please check the build log ""$buildLogFilePath"" for details." }
```

Comment:
Since line 292 declares the variable as a boolean, it is not possible for `$buildSucceeded` to be a $null.

Recommendation: refactor so that you have a single `$buildFailed` variable on line 292. Move the `throw` statements from lines 302-203 to come next, so that you immediately throw an error `if ($buildFailed)`. Follow that with lines 298 and 299. No need for an "if succeeded" type statement.

Reply:
Updated